src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/FormatTask.kt

```diff
@@ -26,26 +28,40 @@ open class FormatTask @Inject constructor(
     @Optional
     val report: RegularFileProperty = objectFactory.fileProperty()
 
+    @Classpath
+    val ktlintClasspath: ConfigurableFileCollection = objectFactory.fileCollection()
+
+    @Classpath
+    val ruleSetsClasspath: ConfigurableFileCollection = objectFactory.fileCollection()
+
     init {
         outputs.upToDateWhen { false }
     }
 
     @TaskAction
     fun run() {
-        val result = with(workerExecutor.noIsolation()) {
-            submit(FormatWorkerAction::class.java) { p ->
-                p.name.set(name)
-                p.files.from(source)
-                p.projectDirectory.set(projectLayout.projectDirectory.asFile)
-                p.ktLintParams.set(getKtLintParams())
-                p.output.set(report)
+        val workQueue = workerExecutor.processIsolation { spec ->
```

Comment:
Are you sure this isolation level works? I remember trying various ones in order to try to get a classloader isolation for the version of kotlin used by ktlint (because it uses internal APIs) to wall it off from the version the user may have configured, but in the end I had a hard time getting it working.
If it does work, is there any performance impact?